File: packages/shared/src/dtos/user.dto.ts
import { z } from 'zod';

export const userRoleSchema = z.enum(['admin', 'lawyer', 'secretary', 'paralegal']);

export const createUserSchema = z.object({
    email: z.string().email('Email inválido'),
    fullName: z.string().min(2, 'Nome é obrigatório'),
    role: userRoleSchema.default('lawyer'),
    oabNumber: z.string().optional(),
});

export const updateUserSchema = createUserSchema.partial();

export type UserRole = z.infer<typeof userRoleSchema>;
export type CreateUserDto = z.infer<typeof createUserSchema>;
export type UpdateUserDto = z.infer<typeof updateUserSchema>;

export interface UserDto {
    id: string;
    tenantId: string;
    email: string;
    fullName: string;
    role: UserRole;
    oabNumber?: string;
    avatarUrl?: string;
    createdAt: string;
    updatedAt: string;
}

File: apps/api/JurisNexo.Application/DTOs/UserDto.cs
namespace JurisNexo.Application.DTOs;

public record UserDto(
    Guid Id,
    Guid TenantId,
    string Email,
    string Name,
    string Role,
    string? AvatarUrl,
    string? Phone,
    bool IsEmailVerified,
    DateTime CreatedAt,
    DateTime UpdatedAt
);
